node-webservice/react-client/src/Components/GameplayPage.jsx

```diff
@@ -161,19 +145,57 @@ class GameplayPage extends Component {
         <div className='gameplay-page'>
           <h3>Playing game {this.props.match.params.gameid}</h3>
           <Grid onCellClick={this.handleCellClicked} parts={this.state.parts}></Grid>
+          <button onClick={this.handleSubmit}> Submit </button>
+          <RobotJSONObjectForm />
         </div>
       );
       //return <RobotJSONObjectForm />;
     }
     if (this.state.gameplayPhase === 'battle') {
+
       return (
-        <div className='gameplay-page'>
-          <h3>Battle!</h3>
-          <Grid onCellClick={ () => {} } parts={this.state.parts}></Grid>          
+        <div className='surrounding-square'>
+          <Grid onCellClick={() => { }} parts={this.state.parts} gameplayPhase={this.state.gameplayPhase}></Grid>
+          <BehaviourBar clicked={this.changeBehaviour} />
+          {this.renderBehaviourText()}
         </div>
       );
     }
-    return <ErrorPage>No page defined for game state: {this.state.gameplayPhase}</ErrorPage> 
+    return <ErrorPage>No page defined for game state: {this.state.gameplayPhase}</ErrorPage>
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    try {
+      socket.emit(
+        'game-message',
+        { action: "submitrobot", parts: this.state.parts, username: this.state.username },
+        response => {
+          if (response.error) {
+            alert("Error processing robot data: " + response.error);
+          }
+        }
+      );
+      //alert("Sent robot data");
+      this.setState({ gameplayPhase: "battle" })
+    } catch (e) {
+      alert("Error sending robot data: " + e);
+    }
+  }
+
+  changeBehaviour(behaviour) {
+    const behaviourMessage = {
+      action: "changeBehaviour",
+      behaviour: behaviour.toLowerCase()
+    }
+    this.setState({behaviour: behaviour});
+    socket.emit(
+      'game-message', behaviourMessage, response => {
+        if (response.error) {
+          alert("Error processing robot data: " + response.error);
```

Comment:
This error message is the same as another one used for something else, it would be good to make this one more specific